### PR TITLE
feat: Add confirm quote request dialog component

### PR DESCRIPTION
**Pull request by JedidiahAwuku**

### CODE REVIEW

File: feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts [hunk 1]
```diff
@@ -60,6 +60,16 @@ export const commerceQuotes = {
       newCart: 'New Cart',
       quotes: 'Quotes',
     },
+    confirmSubmitDialog: {
+      title: 'Confirm Send Quote 000000123?',
```
Comment (cawmichae) [inline feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts]: Should the quote be hard coded?

Should this be i18n alised?

File: feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts [hunk 2]
```diff
@@ -60,6 +60,16 @@ export const commerceQuotes = {
...
+      name: { heading: 'Name:', content: 'Quote 00000275' },
+      description: { heading: 'Description:' },
+
+      warningSubmitNote:
+        'Once a request for quote is submitted it cannot be modified.',
+      confirmSubmitNote: 'Are you sure you want to submit this quote?',
```
Comment (cawmichae) [inline feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts:70]: i18n?

File: feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts (hunk 1 above)
Comment (JedidiahAwuku) [reply feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts]: Resolved now. Thanks

File: feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts (hunk 2 above)
Comment (JedidiahAwuku) [reply feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts:70]: Also resolved. Thanks

File: feature-libs/commerce-quotes/components/commerce-quotes-actions-by-role/commerce-quotes-actions-by-role.component.ts [hunk 3]
```diff
@@ -4,28 +4,79 @@
...
+        this.launchDialogService.dialogClose
+          .pipe(
+            filter((reason) => reason === 'yes'),
+            tap(() =>
+              this.commerceQuotesService.performQuoteAction(quoteCode, action)
+            )
+          )
+          .subscribe()
+      );
+    } else {
```
Comment (cawmichae) [inline feature-libs/commerce-quotes/components/commerce-quotes-actions-by-role/commerce-quotes-actions-by-role.component.ts]: Could put a return statement at the end of the if block, and so remove the else

Comment (JedidiahAwuku) [reply feature-libs/commerce-quotes/components/commerce-quotes-actions-by-role/commerce-quotes-actions-by-role.component.ts]: Done. Thanks

**Review by cawmichae — APPROVED** [review]

**Pull request merged**